docs/preview/03-Features/secret-store/create-new-secret-provider.md

```diff
@@ -181,8 +165,48 @@ namespace Application.Controllers
 }
 ```
 
-### Adding secret name mutation before looking up secret
+### Adding secret versions to your secret provider
+When your secret storage location supports secret versions, you could consider adapting your secret provider to support these secret versions.
+For more information on how you can use many secret versions in your application, see [this dedicated page](./versioned-secret-provider.md).
+
+Implement from `IVersionedSecretProvider` instead of `ISecretProvider` to allow the secret store to pick that your secret provider supports secret versions.
+The following example shows how the registry secret provider only supports two versions of a secret:
+```csharp
+using Arcus.Security.Core;
+using Microsoft.Win32;
+
+public class RegistrySecretProvider : IVersionedSecretProvider
+{
+    // Also implement the general `ISecretProvider` methods...
+
+     public Task<IEnumerable<string>> GetRawSecretsAsync(string secretName, int amountOfVersions)
```

Comment:
So, you retrieve actually a 'history' of secrets .   Wouldn't it be better that you can define:  Hey, I want version X of this secret ?

Reply:
Hmm, I might get the idea.  You want to get for instance the last 3 values of a certain secret, and if the most recent version doesn't work, you fall back on the previous one ?

Reply:
> So, you retrieve actually a 'history' of secrets . Wouldn't it be better that you can define: Hey, I want version X of this secret ?

That was not the initial reason why this feature was introduced. The purpose was to make sure that multiple versions of a single secret (name) were able to be retrieved from the secret store. This would allow (for example) easier key rotation in Key Vault while still supporting 'older' versions, up to the one the consumer defines.
From a user perspective, (if the secret versions are generated, for example), that would be an implementation detail and of no use to the actual secret value.

> Hmm, I might get the idea. You want to get for instance the last 3 values of a certain secret, and if the most recent version doesn't work, you fall back on the previous one ?

Yes, implementations that uses the secret store and use it like that. Depends on how the secret store is used and application specifics, of course.

Reply:
Yes, I agree.  Because typically, consumers don't care about the version-number.   Read the issue / feat request as well in the meantime :)